mergekit/plan.py

```diff
@@ -85,16 +84,36 @@ def __init__(
         if config.base_model and config.align_weights:
             self._space_planner = SpacePlanner(config.base_model)
 
-    @lru_cache
-    def model_arch_info(self, model: ModelReference):
-        return ConfiguredArchitectureInfo(
-            info=self.arch_info,
-            config=model.config(trust_remote_code=self.trust_remote_code),
-        )
+        self.arch_dict: Dict[ModelReference, ConfiguredArchitectureInfo] = {}
+        _models = config.referenced_models()
+        base_model = _models[0]
+        base_config = base_model.config(trust_remote_code=options.trust_remote_code)
+        self.arch_info = get_architecture_info(base_config)
+
+        for m in config.referenced_models():
+            m_config = m.config(trust_remote_code=options.trust_remote_code)
+            configured_arch_info = ConfiguredArchitectureInfo(
+                info=get_architecture_info(m_config),
+                config=m.config(m_config),
+            )
 
-    def normalize_config(self):
-        base_model = self.config.base_model
+            if config.align_weights:
+                mapping = None
+                for arch, destination_arch in [
+                    (arch1, arch2)
+                    for arch1 in base_config.architectures
+                    for arch2 in m_config.architectures
+                ]:
+                    mapping = JSON_MAPPINGS.get(arch, {}).get(destination_arch)
```

Comment:
It'd probably be good to throw an error in this part if the architecture pair isn't supported - better to let people know it's not expected to work than fail with a much more obtuse error later in the process.